Blake/rna 389 validate all usages of rnc sequence regions in the piepline

blakesweeney opened the pull request:
I think this updates all queries we want only the active coordinates from querying `rnc_sequence_regions` to `rnc_sequence_regions_active`. Not sure if I missed anything, but hopefully it doesn't cause any problems. I also delete a couple files which I think are unused.

blakesweeney committed:
Remove unused query

This query is years old and not used anywhere. It should be removed to
make things clearer.

blakesweeney committed:
Use active coordinates when precomputing

blakesweeney committed:
Delete unused file

As far as I can tell this file is not used anywhere, so I am removing
it.

blakesweeney committed:
Use only active to determine what to map

This uses the rnc_sequence_regions_active table to determine what to
map. This is correct as it will try to map something if the accession is
obsoleted.

blakesweeney committed:
Switch to using a materialized view

I want to use one of these since it is safer. No one can accidentally use
it to insert data and it makes clear this is a computed result. We can
still index it so we get high performance.

afg1 reviewed (commented):
I think we need this change in a couple more places:
- `files/crs/fetch-rfam.sql` unless precompute `is_active` will catch it?
- `files/crs/fetch-rnacentral.sql` same, if precompute `is_active` catches it, no worry
- `files/ftp-export/genome_coordinates/query.sql` 
- `files/import-data/evlncrnas/dump-lookup.sql`
- `files/import-data/post-release/002__Cleanup_assembly_table.sql` assuming we don't want assemblies with no active regions?

This one I'm not so sure about:
- `files/repeats/find-assemblies.sql`

And it looks like the table is referenced from some python which I should probably change:
- `rnacentral_pipeline/databases/evlncrnas/lookup.py`

blakesweeney committed:
Correct usage of rnc_sequence_regions in CRS import

Not too sure if this is needed, but looking at the import logic this
can't hurt.

blakesweeney committed:
Remove commented out code

I assume it isn't used anymore.

blakesweeney commented:
- `files/crs/fetch-rfam.sql` and `files/crs/fetch-rnacentral.sql`, not sure they are needed or not. I made the change anyway. We haven't imported CRS in a while, maybe we should fix that, so I think it is ok.
- I think I already changed `files/ftp-export/genome_coordinates/query.sql`. 
- `files/import-data/evlncrnas/dump-lookup.sql` has a lot of commented out code where it appears. I removed the commented out code.
- `files/import-data/post-release/002__Cleanup_assembly_table.sql` - I'm going to leave this one. I think we should stop removing assemblies like this and you have done some work to allow multiple assemblies per species.

The repeats is fine, this isn't used but needs to be finished properly.

The evlncrna code probably shouldn't fetch regions. We generally don't import regions from the databases which are imported by reference. Also, that file has a bunch of commented out code that should be removed.
